File: src/API/Modules/Movies/Cinemanjaro.Movies.Core/Entities/Movie.cs
using Cinemanjaro.Movies.Core.Enums;
using MongoDB.Bson;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cinemanjaro.Movies.Core.Entities
{
    public class Movie
    {
        public ObjectId Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public IEnumerable<string> Directors { get; set; }
        public IEnumerable<string> Writers { get; set; }
        public IEnumerable<string> Actors { get; set; }
        public IEnumerable<Genre> Genres { get; set; }
        public int LengthMins { get; set; }
        public double Rate { get; set; }
        public int ReleaseYear { get; set; }
        public string IMDBLink { get; set; }
        public string FilmwebLink { get; set; }
    }
}

File: src/APP/cinemanjaro-app/src/app/models/Movie.ts
export interface Movie {
    id: string,
    title: string,
    description: string,
    directors: string[],
    actors: string[],
    genres: string[],
    writers: string[],
    lengthMins: 0,
    rate: 0,
    releaseYear: 0,
    imdbLink: string,
    photoURL: string,
}

export interface MoviesResponse{
    movies:MovieListElementData[]
}

export interface MovieListElementData{
    id:string,
    iconURL:string,
    genres:string[],
    title:string
}

export interface AllMoviesResponse{
    movies:AllMoviesListElementData[]
    amount:number
}

export interface AllMoviesListElementData{
    id:string,
    title:string,
    photoURL:string
}